rllib/examples/env/cartpole_crashing.py

```diff
@@ -0,0 +1,72 @@
+import logging
+from gym.envs.classic_control import CartPoleEnv
+import numpy as np
+import time
+
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.error import EnvError
+
+logger = logging.getLogger(__name__)
+
+
+class CartPoleCrashing(CartPoleEnv):
+    """A CartPole env that crashes from time to time.
+
+    Useful for testing faulty sub-env (within a vectorized env) handling by
+    RolloutWorkers.
+
+    After crashing, the env expects a `reset()` call next (calling `step()` will
+    result in yet another error), which may or may not take a very long time to
+    complete. This simulates the env having to reinitialize some sub-processes, e.g.
+    an external connection.
+    """
+
+    def __init__(self, config=None):
+        super().__init__()
+
+        config = config or {}
+
+        # Crash probability (in each `step()`).
+        self.p_crash = config.get("p_crash", 0.005)
```

Comment:
if you want to have finer control over when an env will crash, you can use:
```
maybe_crash_every_n_steps = n
p_crash_per_step = p
```
then, a random crashing env will have:
```
maybe_crash_every_n_steps = 1
p_crash_per_step = 0.005
```
and an env crashing at specific step n will have:
```
maybe_crash_every_n_steps = n
p_crash_per_step = 1.0
```

Reply:
done